### PR TITLE
feat: implement getPushDeviceSettingsList

### PR DESCRIPTION
**Pull request by alexruzenhack**
### Acceptance Criteria
- implement getPushDeviceSettingsList -- it will feed `onTxPushNotificationRequested` lambda.

### Security Checklist
- [ ] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: tests/db.test.ts [hunk 1]
```diff
@@ -2640,3 +2642,207 @@ describe('getPushDevice', () => {
...
+        enablePush: false,
+        enableShowAmounts: false,
+      },
+    ];
+
+    // register wallets
+    const loadWallet = (eachDevice) => createWallet(mysql, eachDevice.walletId, XPUBKEY, AUTH_XPUBKEY, 5);
+    await devicesToLoad.forEach(loadWallet);
+
+    // regiter devices
```
Comment (andreabadesso) [inline tests/db.test.ts]: register*

**Review by andreabadesso — APPROVED** [review]

File: src/db/index.ts [hunk 2]
```diff
@@ -2904,3 +2905,35 @@ export const getPushDevice = async (
...
+    SELECT wallet_id
+         , device_id
+         , enable_push
+         , enable_show_amounts
+      FROM \`push_devices\`
+     WHERE wallet_id in (?)`,
+    [walletIdList],
+  ) as Array<{wallet_id, device_id, enable_push, enable_show_amounts}>;
+
+  const pushDeviceSettignsList = pushDeviceSettingsResult.map((each) => ({
```
Comment (tuliomir) [inline src/db/index.ts]:
```suggestion
  const pushDeviceSettingsList = pushDeviceSettingsResult.map((each) => ({
```

File: tests/db.test.ts [hunk 3]
```diff
@@ -2640,3 +2642,207 @@ describe('getPushDevice', () => {
...
+
+    // devices
+    const devicesToLoad = deviceCandidates.filter((each) => each.enablePush === true);
+    const devicesToNotLoad = deviceCandidates.filter((each) => each.enablePush === false);
+
+    // register wallets
+    const loadWallet = (eachDevice) => createWallet(mysql, eachDevice.walletId, XPUBKEY, AUTH_XPUBKEY, 5);
+    await devicesToLoad.forEach(loadWallet);
+
+    // regiter devices
```
Comment (tuliomir) [inline tests/db.test.ts]:
```suggestion
    // register devices
```

File: tests/db.test.ts [hunk 4]
```diff
@@ -2640,3 +2642,207 @@ describe('getPushDevice', () => {
...
+      enableShowAmounts: eachDevice.enableShowAmounts,
+    });
+    await devicesToLoad.forEach(loadDevice);
+
+    // get settings
+    const notRegisteredWalletIdList = devicesToNotLoad.map((each) => each.walletId);
+    const result = await getPushDeviceSettingsList(mysql, notRegisteredWalletIdList);
+
+    // assert settings
+    expect(result).toStrictEqual([]);
```
Comment (tuliomir) [inline tests/db.test.ts:2706]: I had some difficulty understanding this test. Maybe more descriptive comments would help future developers understand it better.

Maybe something along the lines of: _"loading only two wallets and their devices on the database, but querying for the other two that were not loaded"_

File: tests/db.test.ts [hunk 5]
```diff
@@ -2640,3 +2642,207 @@ describe('getPushDevice', () => {
...
+
+    // devices
+    const devicesToLoad = deviceCandidates.filter((each) => each.enablePush === true);
+    const devicesToNotLoad = deviceCandidates.filter((each) => each.enablePush === false);
+
+    // register wallets
+    const loadWallet = (eachDevice) => createWallet(mysql, eachDevice.walletId, XPUBKEY, AUTH_XPUBKEY, 5);
+    await devicesToLoad.forEach(loadWallet);
+
+    // regiter devices
```
Comment (tuliomir) [inline tests/db.test.ts]:
```suggestion
    // register devices
```

File: tests/db.test.ts [hunk 6]
```diff
@@ -2640,3 +2642,207 @@ describe('getPushDevice', () => {
...
+    await devicesToLoad.forEach(loadDevice);
+
+    // get settings
+    const walletIdList = deviceCandidates.map((each) => each.walletId);
+    const result = await getPushDeviceSettingsList(mysql, walletIdList);
+
+    // assert settings
+    expect(result).toHaveLength(2);
+
+    // verify devices not loaded, they should yeild and empty list
```
Comment (tuliomir) [inline tests/db.test.ts]:
```suggestion
    // verify devices not loaded, they should yield an empty list
```

**Review by tuliomir — APPROVED** [review]

**Pull request merged**